### PR TITLE
PXP-630: [CLI]: fix ctrl+e panic

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
Sometimes when pressing the Ctrl-e hotkey, the CLI went to panic because the `calculate_start_position` function can result in overflow when `state.logs_table_start_position `exceeds the length of state.log_entries.

Ticket: https://mindvalley.atlassian.net/browse/PXP-630

### Fixed

- [x] Replaced the direct subtraction of state.logs_table_start_position with the saturating_sub method to ensure that the value stays within a safe range.

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/165#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — APPROVED** [review]
LGTM

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/165#pullrequestreview-1710549644" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/165#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
